ui/app/peers/create/[peerType]/helpers/my.ts

```diff
@@ -100,6 +105,117 @@ export const mysqlSetting: PeerSetting[] = [
     optional: true,
     tips: 'If not provided, host CA roots will be used.',
   },
+
+  {
+    label: 'Authentication type',
+    field: 'authType',
+    default: 'MYSQL_PASSWORD',
+    stateHandler: (value, setter) =>
+      setter((curr) => ({
+        ...(curr as MySqlConfig),
+        authType: mySqlAuthTypeFromJSON(value),
+      })),
+    type: 'select',
+    placeholder: 'Select authentication mechanism',
+    options: [
+      { value: 'MYSQL_PASSWORD', label: 'Password' },
+      { value: 'MYSQL_IAM_AUTH', label: 'AWS IAM Auth' },
+    ],
+    tips: 'AWS IAM Auth is supported for all Mirror Types for MySQL',
+  },
+  {
+    label: 'AWS IAM Auth Mechanism',
+    field: 'awsAuth.authType',
+    default: 'IAM_AUTH_AUTOMATIC',
+    stateHandler: (value, setter) =>
+      setter((curr) => {
+        let cfg = curr as MySqlConfig;
+        let awsAuth: AwsAuthenticationConfig = {
+          region: '',
+          authType: awsIAMAuthConfigTypeFromJSON(value),
+        };
+        switch (awsAuth.authType) {
+          case AwsIAMAuthConfigType.IAM_AUTH_STATIC_CREDENTIALS:
+            awsAuth = {
+              ...awsAuth,
+              staticCredentials: {
+                accessKeyId: '',
+                secretAccessKey: '',
+              },
+            };
+            break;
+          case AwsIAMAuthConfigType.IAM_AUTH_ASSUME_ROLE:
+            awsAuth = {
+              ...awsAuth,
+              role: {
+                assumeRoleArn: '',
```

Comment:
should reset `chainedIamRole` too?

Reply:
chainedIamRole is nullable so I think setting the `role` object should do it?